dvclive/plots.py

```diff
@@ -0,0 +1,102 @@
+import json
+
+import numpy as np
+from sklearn import calibration, metrics
+
+
+def _check_binary(y_true):
+    if len(np.unique(y_true)) > 2:
+        raise ValueError(
+            "Only binary classification is supported for roc_curve."
+        )
+
+
+def _dump(content, output_file):
+    with open(output_file, "w") as f:
+        json.dump(content, f, indent=4)
+
+
+def log_roc(y_true, y_score, output_file, **kwargs):
+    _check_binary(y_true)
+
+    fpr, tpr, roc_thresholds = metrics.roc_curve(
+        y_true=y_true, y_score=y_score, **kwargs
+    )
+
+    roc = {
+        "roc": [
+            {"fpr": fp, "tpr": tp, "threshold": t}
+            for fp, tp, t in zip(fpr, tpr, roc_thresholds)
+        ]
+    }
+
+    _dump(roc, output_file)
+
+    return roc
+
+
+def log_precision_recall(y_true, probas_pred, output_file, **kwargs):
+    _check_binary(y_true)
+
+    precision, recall, prc_thresholds = metrics.precision_recall_curve(
+        y_true=y_true, probas_pred=probas_pred, **kwargs
+    )
+
+    prc = {
+        "prc": [
+            {"precision": p, "recall": r, "threshold": t}
+            for p, r, t in zip(precision, recall, prc_thresholds)
+        ]
+    }
+    _dump(prc, output_file)
+
+    return prc
+
+
+def log_confusion_matrix(y_true, y_pred, output_file):
```

Comment:
Why no `**kwargs` here?

Reply:
This is the only case where we are not calling `sklearn` internal function but just reformating the data into DVC format . There were kwargs in previous version because we were calling `sklearn.metrics.confusion_matrix` and saving the plot as image.